src/backend/distributed/commands/utility_hook.c

```diff
@@ -744,6 +743,64 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 }
 
 
+/*
+ * CreateCustomDDLTaskList creates a DDLJob which will apply a command to all placements
+ * of shards of a distributed table. The command to be applied is generated by the
+ * GenerateDDLCommandForShardFn function passed in. The generator receives the fully
+ * qualified shardName to apply the command on. A void * is passed as context to the
+ * called function.
+ */
+DDLJob *
+CreateCustomDDLTaskList(Oid relationId, GenerateDDLCommandForShardFn fn,
```

Comment:
It's mildly confusing that this uses a function API that's different from the ones used for makeTableDDLCommandFunction. 

We're doing AppendShardIdToName in this function, but also in GetShardedTableDDLCommandColumnar. I'm wondering whether we could do it one place.

Reply:
Good point. This function predates the other infra structure. Let me see if there is an easy way I can combine both.